File: client/src/types/quiz.ts
// Quiz
export type Quiz = {
  id: string;
  name: string;
  description: string;
  questions: QuizQuestion[];
}

// QuizQuestion
export type QuizQuestion = {
  id: string;
  name: string;
  choices: QuizChoice[];
}

// QuizChoice
export type QuizChoice = {
  id: string;
  name: string;
  correct: boolean;
}

export type Player = {
  id: string;
  name: string;
}

File: internal/entity/quiz.go
package entity

type Quiz struct {
	Id string `json:"id"`
	Name string `json:"name"`
	Questions []QuizQuestion `json:"questions"`
}

type QuizQuestion struct {
	Id string `json:"id"`
	Name string `json:"name"`
	Choices []QuizChoice `json:"choices"`
}

type QuizChoice struct {
	Id string `json:"id"`
	Name string `json:"name"`
	Correct bool `json:"correct"`
}